camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelEndpointAnnotator.java

```diff
@@ -105,6 +114,25 @@ void validateText(@NotNull PsiElement element, @NotNull AnnotationHolder holder,
         }
     }
 
+    private void validateEndpointReference(PsiElement element, String camelQuery, AnnotationHolder holder) {
+        if (!getIdeaUtils().isJavaLanguage(element)) { //no need, unresolvable references in XML are already highlighted
+            return;
+        }
+        if (CamelEndpoint.isDirectEndpoint(camelQuery)) { //only direct endpoints have references (for now)
+            DirectEndpointReference endpointReference = Arrays.stream(element.getReferences())
+                .filter(r -> r instanceof DirectEndpointReference)
+                .map(r -> (DirectEndpointReference) r)
+                .findAny().orElse(null);
+            if (endpointReference != null) {
```

Comment:
if you add a `.map`after findAny then you might be able to handle the inside of `if (endpointReference != null)`and get rid of the if statement.

I haven't checked if any of the variable scope needs to be final.